app/decoding/communities/[communityId]/page.tsx

```diff
@@ -0,0 +1,1451 @@
+    )
+
+    return (
+        <div className="space-y-6">
+            {/* Simple Filter Tabs */}
+            <div className="flex items-center gap-2 border-b border-gray-200 dark:border-gray-800">
+                {resourceTypes.map((type) => (
+                    <button
+                        key={type.value}
+                        onClick={() =>
+                            setSelectedType(type.value as ResourceType | "all")
+                        }
+                        className={`
+                            px-4 py-2 text-sm font-medium relative
+                            ${
+                                selectedType === type.value
+                                    ? "text-orange-500 dark:text-orange-400"
+                                    : "text-gray-600 dark:text-gray-400 hover:text-gray-900 dark:hover:text-gray-300"
+                            }
+                        `}
+                    >
+                        {type.label}
+                        {selectedType === type.value && (
+                            <div className="absolute bottom-0 left-0 right-0 h-0.5 bg-orange-500 dark:bg-orange-400" />
+                        )}
+                    </button>
+                ))}
+            </div>
+
+            {/* Resources Grid */}
+            <div className="grid gap-4">
+                {filteredResources.map((resource) => (
+                    <ResourceCard key={resource.id} resource={resource} />
+                ))}
+            </div>
+
+            {/* Simple Empty State */}
+            {filteredResources.length === 0 && (
+                <div className="text-center py-8">
+                    <p className="text-gray-600 dark:text-gray-400">
+                        No {selectedType === "all" ? "resources" : selectedType}{" "}
+                        available for this week.
+                    </p>
+                </div>
+            )}
+        </div>
+    )
+}
+
+export default function CommunityDashboard({
+    params
+}: {
+    params: { communityId: string }
+}) {
+    const [communityData, setCommunityData] = useState<Community | null>(null)
+    const [selectedWeek, setSelectedWeek] = useState(1)
+    const [currentWeekData, setCurrentWeekData] = useState<WeeklyData | null>(
+        null
+    )
+    const [activeTab, setActiveTab] = useState("Learning Path")
+
+    const isCompleted = communityData ? isCohortCompleted(communityData) : false
+
+    useEffect(() => {
+        const community = getCommunity(params.communityId)
+        if (community) {
+            setCommunityData(community)
+            setSelectedWeek(getCurrentWeek(community))
+        }
+    }, [params.communityId])
+
+    useEffect(() => {
+        if (communityData) {
+            const weekData = getWeeklyData(communityData, selectedWeek)
+            setCurrentWeekData(weekData || null)
+        }
+    }, [selectedWeek, communityData])
+
+    if (!communityData || !currentWeekData) {
```

Comment:
This page would be better suited as a server component.
So we generate all possible community pages on build and then it is easy to throw not found for community routes that do not exist.
the current approach returns "Loading"

also with the page rendered as a server component we get the data on the server and then pass them to client/server components.
Removes the need for useState and useEffect to store communityData and currentWeekData